tornado/ioloop.py

```diff
@@ -945,18 +945,21 @@ def start(self):
                 self._events.update(event_pairs)
                 while self._events:
                     fd, events = self._events.popitem()
-                    try:
-                        fd_obj, handler_func = self._handlers[fd]
-                        handler_func(fd_obj, events)
-                    except (OSError, IOError) as e:
-                        if errno_from_exception(e) == errno.EPIPE:
-                            # Happens when the client closes the connection
-                            pass
-                        else:
-                            self.handle_callback_exception(self._handlers.get(fd))
-                    except Exception:
-                        self.handle_callback_exception(self._handlers.get(fd))
-                fd_obj = handler_func = None
+                    h = self._handlers.get(fd)
+                    # Another callback may have removed the handler
```

Comment:
I don't see how this could happen. Whenever we remove from `self._handlers`, we also remove from `self._events` (https://github.com/tornadoweb/tornado/blob/3163c606f06065b296df70bef5390afa548825d8/tornado/ioloop.py#L797)